BDArmory/Control/MissileFire.cs

```diff
@@ -6143,8 +6143,17 @@ public void SendTargetDataToMissile(MissileBase ml, bool clearHeat = true)
                 if (BDArmorySettings.DEBUG_MISSILES)
                     Debug.Log($"[BDArmory.MissileData]: targetInfo sent for {ml.targetVessel.Vessel.GetName()}");
             }
+            else if (ml.TargetingMode == MissileBase.TargetingModes.Heat && ml.hasDataLink && !heatTarget.exists)
+            {
+                if (vesselRadarData)
+                {
+                    ml.vrd = vesselRadarData;
+                    if (vesselRadarData.locked) ml.targetVessel = vesselRadarData.lockedTargetData.targetData.vessel.gameObject.GetComponent<TargetInfo>();
+                }
+            }
```

Comment:
Heaters' only use of radar is to potentially do uncaged locks to tell the seeker head where to point; more than that, and you've turned the heatseeker into a radar missile. At this point, you'd be better off with a radar guided missile with heat terminal guidance.

Reply:
The idea here is do something similar to some modern heatseekers (MICA IR, Type 81 SAM) that can guide the Missile outside it's Max seeker Lock range via DataLink and Inertial Navigation and after the missile gets a solid lock it close it's connection and act as a normal heat seeker , my idea here is a full heat seeker that can be used in planes without radar that can be fired and guided by radar if the the target is inside missile range and not inside seeker range, and after it gets a lock it disable datalink (ln 755 on MissileBase)